java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/VxlanNI.java

```diff
@@ -0,0 +1,108 @@
+package org.projectfloodlight.openflow.types;
+
+import java.util.Arrays;
+
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.projectfloodlight.openflow.exceptions.OFParseError;
+
+import com.google.common.hash.PrimitiveSink;
+import com.google.common.primitives.UnsignedInts;
+
+/** Represents the VXLAN Network Identifier (24 bits).
+ *
+ * @author Sarath Kumar Sankaran Kutty
+ * {@literal <}sarath.kutty@bigswitch.com{@literal >}
+ */
+public class VxlanNI implements OFValueType<VxlanNI> {
+
+    private static final int VALIDATION_MASK = 0x00FFffFF;
+    private static final int ZERO_VALUE = 0x0;
+    final static int LENGTH = 3;
+
+    public static final VxlanNI ZERO_INVALID = new VxlanNI(ZERO_VALUE);
+
+    private final int vni;
+
+    private VxlanNI(int vni) {
+        this.vni = vni;
+    }
+
+    public static VxlanNI ofVni(int vni) {
+        if (vni == ZERO_INVALID.vni || (vni & VALIDATION_MASK) != vni) {
+            throw new IllegalArgumentException(String.format("Illegal VNI value: %x", vni));
+        }
+        return new VxlanNI(vni);
+    }
+
+    /** @return the VxLAN Network ID */
+    public int getVni() {
+        return vni;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null) return false;
+        if (getClass() != obj.getClass()) return false;
+        VxlanNI other = (VxlanNI) obj;
+        if (vni != other.vni) return false;
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + vni;
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "VxlanNI [vni=" + vni + "]";
+    }
+
+    @Override
+    public int getLength() {
+        return LENGTH;
+    }
+
+    private volatile byte[] bytesCache = null;
```

Comment:
since the VNI is pretty short (3 bytes) you don't need the cache etc. On every `getBytes()` call you can just allocate a new byte array and fill it from the VNI.